src/pages/movies/[id].tsx

```diff
@@ -1,30 +1,20 @@
 import {GetServerSideProps, NextPage} from "next"
-import {Data} from "./index"
 import {Movie} from "../../types"
 
 
 
-/* export const getServerSideProps: GetServerSideProps = async (context) => {
-    const res = await fetch(`http://localhost:3000/api/movies/${context.params!.id}`)
+export const getServerSideProps: GetServerSideProps = async (context) => {
+    const protocol = context.req.headers["x-forwarded-proto"] || "http";
+    const baseUrl = context.req ? `${protocol}://${context.req.headers.host}` : "";
+    const res = await fetch(`${baseUrl}/api/movies/${context.params!.id}`)
```

Comment:
I actually like this solution. It will work in any environment, without any manual configuration. The only downside is that it's a lot of code, but that could be solved by separating it out into a utility function, and use that everywhere (like here).